File: app/Global.java
import com.google.inject.Guice;
import com.google.inject.Injector;
import hubs.GuiceDependencyResolver;
import play.Application;
import play.api.mvc.EssentialFilter;
import signalJ.GlobalHost;
import signalJ.SignalJGlobal;
import signalJ.infrastructure.CORSFilter;
import signalJ.infrastructure.CorsPolicy;
import signalJ.infrastructure.impl.AllowAllCorsPolicy;

public class Global extends SignalJGlobal {
    final Injector injector = Guice.createInjector(new GuiceModule());

    @Override
    public <A> A getControllerInstance(Class<A> aClass) throws Exception {
        return injector.getInstance(aClass);
    }

    @Override
    public void onStart(Application application) {
        final GuiceDependencyResolver resolver = new GuiceDependencyResolver(injector);
        GlobalHost.setDependencyResolver(resolver);
        final CorsPolicy policy = new AllowAllCorsPolicy();
        GlobalHost.getDependencyResolver().register(CorsPolicy.class, () -> policy);
    }

    @Override
    public <T extends EssentialFilter> Class<T>[] filters() {
        return new Class[] {CORSFilter.class};
    }
}

File: app/Global.scala
import com.google.inject.{Guice, Injector}
import hubs.GuiceDependencyResolver
import play.api.mvc.{EssentialAction, Filters, Handler, RequestHeader}
import play.api.{Application, GlobalSettings}
import signalJ.infrastructure.CORSFilter
import signalJ.{GlobalHost, TransportTransformer}

object Global extends GlobalSettings {
  val injector: Injector  = Guice.createInjector(new GuiceModule())

  override def onRouteRequest(request: RequestHeader): Option[Handler] = {
    super.onRouteRequest(TransportTransformer.transform(request))
  }

  override def getControllerInstance[A](controllerClass: Class[A]): A = {
    injector.getInstance(controllerClass)
  }

  override def onStart(app: Application) {
    //This is an example of setting your own resolver
    val resolver: GuiceDependencyResolver = new GuiceDependencyResolver(injector)
    GlobalHost.setDependencyResolver(resolver)
    /*val policy: CorsPolicy = new AllowAllCorsPolicy
    GlobalHost.getDependencyResolver.Register(classOf[CorsPolicy], () -> policy)*/
  }

  override def doFilter(next: EssentialAction): EssentialAction = {
    Filters(super.doFilter(next), CORSFilter)
  }
}